src/ess/powder/grouping.py

```diff
@@ -12,23 +14,42 @@
 )
 
 
+def _drop_grouping_and_bin(
+    data: sc.DataArray, *, dims_to_reduce: tuple[str, ...] | None = None, edges: dict
+) -> sc.DataArray:
+    all_pixels = data if dims_to_reduce == () else data.bins.concat(dims_to_reduce)
+    # all_pixels may just have a single bin now, which currently yields
+    # inferior performance when binning (no/bad multi-threading?).
+    # We operate on the content buffer for better multi-threaded performance.
+    if all_pixels.ndim == 0:
+        content = all_pixels.bins.constituents['data']
```

Comment:
It seems like this should be done by `bin` automatically. Is this possible?

Reply:
See what I wrote in the PR:

> Bypass the same single-threading problem encountered in SANS, I think we either have to put a helper into ESSreduce, or see if the underlying problem should be fixed in Scipp.

Reply:
Ok. Can this be done in Scipp now instead of adding extra code here and potentially in other projects? If not, please open an issue!

Reply:
Well, the larger issue is that we have to `concat` first, which in itself is avoiding other performance issues. Making many->many mapping is a long-standing issues and has seen a couple improvements in the past, but is apparently not fully solved (See scipp/scipp#1846).

What could *probably* be done is bypass the single-threading issue, but it also required some thought (not sure there aren't any subtleties yet), but it is slightly odd to put an optimization for a solution that bypasses another problem.